insta/src/runtime.rs

```diff
@@ -51,43 +55,86 @@ macro_rules! elog {
 #[derive(Debug)]
 pub struct AutoName;
 
-impl From<AutoName> for ReferenceValue<'static> {
-    fn from(_value: AutoName) -> ReferenceValue<'static> {
-        ReferenceValue::File(None)
+pub struct InlineValue<'a>(pub &'a str);
+
+impl From<AutoName> for SnapshotName<'static> {
+    fn from(_value: AutoName) -> SnapshotName<'static> {
+        None
     }
 }
 
-impl From<Option<String>> for ReferenceValue<'static> {
-    fn from(value: Option<String>) -> ReferenceValue<'static> {
-        ReferenceValue::File(value.map(Cow::Owned))
+type SnapshotName<'a> = Option<Cow<'a, str>>;
+
+pub enum SnapshotValue<'a> {
+    Text {
+        name: SnapshotName<'a>,
+        content: &'a str,
+    },
+
+    InlineText {
+        reference_content: &'a str,
+        content: &'a str,
```

Comment:
Do we ever use both of these fields? Can we add docstrings here too?

Reply:
I've added some docstrings to this enum. Regarding the first question: do you mean use both of them together? They get used separately in `assert_snapshot`. The only reason they're in the same enum is only to prevent the invalid `Inline + Binary` combination. This is basically the first variant of what I had suggested in [this comment](https://github.com/mitsuhiko/insta/pull/489#issuecomment-2168227959). But I'm now strongly considering doing something more like the second variant for more consistency (and probably less code because I could replace all those `From` impls with only one).

(thx for the quick feedback btw)

Reply:
Does the same information go in each field? I'm still a bit unclear what each field is for. (I'm sure it's not rocket science, but I'm also thinking of the next person who comes across the code and wants to understand it...)

Or ofc feel free to just do the refactor if you think that makes it simpler!

Reply:
Ah, is `content:` the generated value? If so, let's use that term in the docstrings.